boa/src/syntax/ast/node/array/mod.rs

```diff
@@ -39,14 +39,27 @@ impl Executable for ArrayDecl {
         for elem in self.as_ref() {
             if let Node::Spread(ref x) = elem {
                 let val = x.run(interpreter)?;
-                let mut vals = interpreter.extract_array_properties(&val).unwrap();
-                elements.append(&mut vals);
-                continue; // Don't push array after spread
+                let iterator_record =
+                    super::super::super::super::builtins::iterable::get_iterator(interpreter, val)?;
+                //not sure what to do with this next_index mentioned in the spec
+                //it is mentioned that it has to be returned from somewhere
+                //https://tc39.es/ecma262/#sec-runtime-semantics-arrayaccumulation
+                //let mut next_index = 0;
```

Comment:
This is basically because we don't follow the spec with arrays. The thing is, arrays have a number of integer-indexed elements, and a `nextIndex`. this `nextIndex` will be the same as `array.length`, and it's defined here: https://tc39.es/ecma262/#sec-array-initializer-runtime-semantics-evaluation

But, due to our array being a vector (which causes performance issues), we don't have this defined. This is partially the reason behind ##816. This means that if we for example create a huge empty array (which should not allocate), say `new Array(50000)`, we will have huge performance issues.

This also means that if we do something like `let a = [5,6,,,6,7,,8];`, we will have an `undefined` value in the non-defined slots, but they should instead be empty slots, and here, `nextIndex` would be 8 (same as length).

Reply:
What do you suggest? Should we leave this comment and commented next_index, or modify the comment, or remove completly?

Reply:
I would change the comment to a `TODO`, explaing this is waiting for a proper internal Array representation.

Reply:
Please check the newly committed rewording.